web3/utils/abi.py

```diff
@@ -1,21 +1,505 @@
+        ...         "type": "function",
+        ...     }
+        ... ]
+        >>> get_function_abi(abi, "multiply", [7, 3])
+        {'constant': False, 'inputs': [{'name': 'a', 'type': 'uint256'}, {\
+'name': 'b', 'type': 'uint256'}], 'name': 'multiply', 'outputs': [{'name': 'result', \
+'type': 'uint256'}], 'payable': False, 'stateMutability': 'nonpayable', \
+'type': 'function'}
+    """
+    if function_identifier is FallbackFn:
+        return get_fallback_function_abi(abi)
+
+    if function_identifier is ReceiveFn:
+        return get_receive_function_abi(abi)
+
+    if function_identifier is None or not is_text(function_identifier):
+        raise TypeError("Unsupported function identifier")
+
+    if abi_codec is None:
+        abi_codec = ABICodec(default_registry)
+
+    args = args or tuple()
+    kwargs = kwargs or dict()
+    num_arguments = len(args) + len(kwargs)
+
+    name_filter = functools.partial(filter_abi_by_name, function_identifier)
+    arg_count_filter = functools.partial(_filter_by_argument_count, num_arguments)
+    encoding_filter = functools.partial(
+        _filter_by_encodability, abi_codec, args, kwargs
+    )
+
+    function_matches = pipe(abi, name_filter, arg_count_filter, encoding_filter)
+    function_candidates: List[Union[ABIFunction, ABIError]] = []
+    for fn in function_matches:
+        if fn["type"] == "error":
+            function_candidates.append(cast(ABIError, fn))
+        elif fn["type"] == "function":
+            function_candidates.append(cast(ABIFunction, fn))
+
+    if len(function_candidates) != 1:
+        matching_identifiers = name_filter(abi)
+        matching_function_signatures = [
+            abi_to_signature(func) for func in matching_identifiers
+        ]
+
+        arg_count_matches = len(arg_count_filter(matching_identifiers))
+        encoding_matches = len(encoding_filter(matching_identifiers))
+
+        error_diagnosis = _mismatched_abi_error_diagnosis(
+            function_identifier,
+            matching_function_signatures,
+            arg_count_matches,
+            encoding_matches,
+            args,
+            kwargs,
+        )
+
+        raise MismatchedABI(error_diagnosis)
+
+    return function_candidates[0]
+
+
+def get_receive_function_abi(contract_abi: ABI) -> ABIReceive:
+    receive_abis = filter_by_type("receive", contract_abi)
+    if receive_abis and receive_abis[0]["type"] == "receive":
+        return cast(ABIReceive, receive_abis[0])
+    else:
+        raise FallbackNotFound("No receive function was found in the contract ABI.")
+
+
+def get_fallback_function_abi(contract_abi: ABI) -> ABIFallback:
+    fallback_abis = filter_by_type("fallback", contract_abi)
+    if fallback_abis and fallback_abis[0]["type"] == "fallback":
+        return cast(ABIFallback, fallback_abis[0])
+    else:
+        raise FallbackNotFound("No fallback function was found in the contract ABI.")
+
+
+def filter_abi_by_name(
```

Comment:
Should this get moved to `eth_utils.abi` as well? Seems like we have `filter_by_type` there.